substrate/frame/staking-async-testing-utils/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Testing utilities for pallet-staking-async.
+//!
+//! This crate provides common functions to setup staking state for testing,
+//! such as bonding validators, nominators, and generating different types of solutions.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+
+use frame_benchmarking::account;
+use frame_support::traits::{fungible::Mutate, Get, LockIdentifier};
+use frame_system::RawOrigin;
+use pallet_staking_async::{
+	ActiveEra, ActiveEraInfo, BalanceOf, Bonded, BondedEras, Config, CurrentEra,
+	ErasStakersOverview, ErasTotalStake, ErasValidatorPrefs, ErasValidatorReward, Ledger,
+	MinValidatorBond, Pallet, RewardDestination, StakingLedger, ValidatorPrefs,
+};
+use sp_runtime::{traits::Zero, Perbill};
+use sp_staking::{EraIndex, PagedExposureMetadata};
+
+// Re-export commonly used types
+pub use pallet_staking_async::asset;
+
+// Re-export core testing utilities from the main pallet
+pub use pallet_staking_async::testing_utils::{
+	clear_validators_and_nominators, create_funded_user, create_stash_controller,
+	create_stash_controller_with_balance, create_unique_stash_controller, create_validators,
+	create_validators_with_nominators_for_era, create_validators_with_seed,
+	migrate_to_old_currency, set_active_era, AccountIdLookupOf,
+};
+
+// STAKING_ID constant
+pub const STAKING_ID: LockIdentifier = *b"staking ";
```

Comment:
Does this have to stay is a leftover? we already have a `mod testing_utils` in `staking-async`.